docs/010_GPG-Git-commits-partially-verified.md

```diff
@@ -0,0 +1,448 @@
+
+What happens when we create a new commit?
+
+When a new commit is created, we need to sign it again with the committer GPG key. That leads to two problems:
+
+1. We can lose the commit signature.
+2. We lose the author's signature.
+
+Let us start with point 1, having not signed commits.
+
+If we re-create new commits using rebase, we lose the original author signature, but we can add the maintainer one, as long as we do it in an environment where the GPG private key is available. For example, if you use the "rebase and merge" button provided by the GitHub UI, the new commit will not be signed because GitHub does not have the committer signing key. You can use a console command where the maintainer GPG private key is available.
+
+That forces you to stop using GitHub UI for merging pull requests if you want all your commits to be signed.
+
+Regarding point 2, losing the author's signature, we first must consider why we do not want to lose it.
+
+We can consider the signature a mechanism to review/accepts changes. If we consider only the maintainers the ultimately responsible for our code, losing the author's signature is not critical, as long as maintainers make sure changes come from a trusted source. In this case, we are not breaking the trust. It is just that we trust maintainers, and they check signatures for other contributors they trust. The maintainers should ensure that commits on a pull request are signed by contributors they trust before merging them.
+
+On the other hand, It would be nice to keep that trust chain transparent so people using a package can check that third-party contributors' changes were accepted without changes.
+
+If you do not want to lose the author's signature, one obvious solution is to merge only using merge commits.
+
+It is out of the scope of this article to describe the pros and cons of losing or not losing the author's signature. A fascinating [article by Mike Gerwitz](https://mikegerwitz.com/2012/05/a-git-horror-story-repository-integrity-with-signed-commits#trust-ensure) explains the different alternatives in detail.
+
+When this happens on GitHub, you see this message on the commit status (if you have vigilant mode enabled):
+
+![Commit with partially verified signature](../media/010/commit-with-partially-verified-signature-on-github.png)
+
+Maybe losing the author's signature is not an issue from the package user or maintainers' point of view. However, from the author's point of view, you can not distinguish anymore between your legitimate commits and the fake ones. Someone else could pretend you have collaborated with their project. It is surprisingly easy to do that. You only need to create a commit and override the author with any users on GitHub.
+
+## Conclusion
+
+When you use signed commits:
+
+- Be aware of the Git limitation of one single signature per commit.
+- Be also aware of what happens to signatures when you use Git commands like "merge" and "rebase".
```

Comment:
Possible new article name: Git Merge and rebase: What happens under the hood

Reply:
OK, but It should be focused on how the command affects the commit signatures.